daemon/algod/api/server/v2/generated/types.go

```diff
@@ -292,274 +408,266 @@ type BuildVersion struct {
 	BlockHeadersCommitment []byte `json:"BlockHeadersCommitment"`
 
-	// The first round the message attests to.
+	// FirstAttestedRound The first round the message attests to.
 	FirstAttestedRound uint64 `json:"FirstAttestedRound"`
 
-	// The last round the message attests to.
+	// LastAttestedRound The last round the message attests to.
 	LastAttestedRound uint64 `json:"LastAttestedRound"`
 
-	// An integer value representing the natural log of the proven weight with 16 bits of precision. This value would be used to verify the next state proof.
+	// LnProvenWeight An integer value representing the natural log of the proven weight with 16 bits of precision. This value would be used to verify the next state proof.
 	LnProvenWeight uint64 `json:"LnProvenWeight"`
 
-	// The vector commitment root of the top N accounts to sign the next StateProof.
+	// VotersCommitment The vector commitment root of the top N accounts to sign the next StateProof.
 	VotersCommitment []byte `json:"VotersCommitment"`
 }
 
-// TealKeyValue defines model for TealKeyValue.
+// TealKeyValue Represents a key-value pair in an application store.
 type TealKeyValue struct {
 	Key string `json:"key"`
 
-	// Represents a TEAL value.
+	// Value Represents a TEAL value.
 	Value TealValue `json:"value"`
 }
 
-// TealKeyValueStore defines model for TealKeyValueStore.
-type TealKeyValueStore []TealKeyValue
+// TealKeyValueStore Represents a key-value store for use in an application.
+type TealKeyValueStore = []TealKeyValue
 
-// TealValue defines model for TealValue.
+// TealValue Represents a TEAL value.
 type TealValue struct {
-
-	// \[tb\] bytes value.
+	// Bytes \[tb\] bytes value.
 	Bytes string `json:"bytes"`
 
-	// \[tt\] value type. Value `1` refers to **bytes**, value `2` refers to **uint**
+	// Type \[tt\] value type. Value `1` refers to **bytes**, value `2` refers to **uint**
 	Type uint64 `json:"type"`
 
-	// \[ui\] uint value.
+	// Uint \[ui\] uint value.
 	Uint uint64 `json:"uint"`
 }
 
-// Version defines model for Version.
+// Version algod version information.
 type Version struct {
 	Build          BuildVersion `json:"build"`
 	GenesisHashB64 []byte       `json:"genesis_hash_b64"`
 	GenesisId      string       `json:"genesis_id"`
 	Versions       []string     `json:"versions"`
 }
 
-// AccountId defines model for account-id.
-type AccountId string
+// AccountID defines model for account-id.
+type AccountID = string
 
 // Address defines model for address.
-type Address string
+type Address = string
 
 // AddressRole defines model for address-role.
 type AddressRole string
 
 // AfterTime defines model for after-time.
-type AfterTime time.Time
+type AfterTime = time.Time
 
-// AssetId defines model for asset-id.
-type AssetId uint64
+// AssetID defines model for asset-id.
+type AssetID uint64
 
 // BeforeTime defines model for before-time.
-type BeforeTime time.Time
+type BeforeTime = time.Time
```

Comment:
This is a change in [the v1.11.0 release](https://github.com/deepmap/oapi-codegen/releases/tag/v1.11.0). You can find a more detailed description of the reasons for the change in there, but here are my takeaways:
* Type aliases are being preferred as opposed to what would previously have been a type definition when a type is a `$ref` to another type.
* They did this to preserve method definitions on the original type (type definitions erase methods whereas aliases "inherit" them), and because working with a lot of type definitions is more cumbersome
* We can opt to disable this behavior with a config options if we want

Reply:
@Eric-Warehime Resolving - Thanks for summarizing your takeaways from _$ref prefers type aliasing_ in https://github.com/deepmap/oapi-codegen/releases/tag/v1.11.0.

* I don't think I have enough experience in Go to have a strongly held view here.  For now, I'm on-board to trust the defaults suggested by _oapi-codegen_.
* And as you said, we can apply `old-aliasing: true` should we later change our mind.

Reply:
Thanks for asking this question, I thought this was a bit odd as well but trusted the project and didn't look into it further until now.

In typical go fashion, I couldn't find much documentation but the project proposal is extremely detailed:
https://golang-id.org/proposal/design/18130-type-alias.html

There is even a higher level article about the motivation:
https://go.dev/talks/2016/refactor.article#TOC_5.1

I agree with sticking to the default option.